lib/orchestrator_client.rb

```diff
@@ -15,31 +17,35 @@ class OrchestratorClient
   def initialize(overrides, load_files=false)
     @config = Config.new(overrides, load_files)
     @config.validate
+    @http_conn = nil
   end
 
   def make_uri(path)
     URI.parse("#{config.root_url}#{path}")
   end
 
   def create_http(uri)
-    http = Net::HTTP.new(uri.host, uri.port)
-    http.use_ssl = true
-    http.ssl_version = :TLSv1_2
-    http.ca_file = config['cacert']
-    http.verify_mode = OpenSSL::SSL::VERIFY_PEER
-    http
+    return @http_conn unless @http_conn.nil?
```

Comment:
This seems to change the semantics. It assumes that we connect to the same host/port pair each time. Maybe that should be moved to the initializer?

Reply:
Mostly this is cleanup: the host/port are pulled from config so it's always the same, but this logic hides that fact.

Reply:
The get/post methods can accept only the path once we've initialized the connection, so I think we should do away with `make_uri` and pull `config.root_url` directly into this method (get rid of passing the `uri`).

Reply:
Okay. I just pushed an update which cleans it up quite a bit and I think addresses what you were asking. Let me know.